lib/ansible/plugins/inventory/docker_machine.py

```diff
@@ -0,0 +1,159 @@
+          required: True
+          choices: ['docker_machine']
+        verbose_output:
+            description: Toggle to (not) include all available nodes metadata (e.g. Image, Region, Size) as a JSON object.
+            type: bool
+            default: yes
+        split_tags:
+          description: for keyed_groups add two variables as if the tag were actually a key value pair separated by a colon, instead of just a single value.
+          required: False
+          type: bool
+          default: no
+        split_separator:
+          description: for keyed_groups when splitting tags this is the separator to split the tag value on.
+          required: False
+          type: str
+          default: ":"
+'''
+
+EXAMPLES = '''
+# Minimal example
+plugin: docker_machine
+
+# Example using constructed features to create groups
+# keyed_groups may be used to create custom groups
+strict: False
+keyed_groups:
+  - prefix: tag
+    key: 'dm_tags'
+
+# Example using tag splitting where the tag is like 'dm_tag_gantry_component:routinator'
+strict: False
+split_tags: True
+split_separator: ":"
+keyed_groups:
+  - prefix: gantry_component
+    key: 'dm_tag_gantry_component'
+'''
+
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_native
+from ansible.module_utils._text import to_text
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+
+import json
+import re
+import subprocess
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+    ''' Host inventory parser for ansible using Docker machine as source. '''
+
+    NAME = 'docker_machine'
+
+    def _run_command(self, *args):
+        return subprocess.check_output(["docker-machine"] + list(args)).decode('utf-8').strip()
+
+    def _populate(self):
+        self.inventory.add_group('all')
+
+        try:
+            self.nodes = self._run_command('ls', '-q').splitlines()
+            for self.node in self.nodes:
+                self.node_attrs = json.loads(self._run_command('inspect', self.node))
+
+                id = self.node_attrs['Driver']['MachineName']
+                self.inventory.add_host(id)
+                self.inventory.add_group(self.node_attrs['DriverName'])
+                self.inventory.add_host(id, group=self.node_attrs['DriverName'])
+                self.inventory.add_host(id, group='all')
+
+                # Find out more about the following variables at: https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html
+                self.inventory.set_variable(id, 'ansible_host', self.node_attrs['Driver']['IPAddress'])
+                self.inventory.set_variable(id, 'ansible_port', self.node_attrs['Driver']['SSHPort'])
+                self.inventory.set_variable(id, 'ansible_user', self.node_attrs['Driver']['SSHUser'])
+                self.inventory.set_variable(id, 'ansible_ssh_common_args', '-o StrictHostKeyChecking=no')
```

Comment:
I'm not sure whether this is a good idea. Maybe you should allow the user to choose between `yes`, `no`, `ask` and `accept-new` (with the default being `accept-new` or `ask`).

Reply:
Good point. I was thinking only of the automation case where there is no user to ask.

Reply:
For automation, `yes` or `accept-new` are probably more secure choices than `no`. In the end, it's up to the user to decide what kind of security requirements they have and which default setting they want.

Reply:
After a bit of experimentation I discovered the same can be achieved with explicit plugin code but instead by use of `compose`, thus I removed the explicit setting of `StrictHostKeyChecking=no` and instead added an example of how the same can be achieved using `compos`.